src/pages/page-names.constants.ts

```diff
@@ -176,3 +178,62 @@ export const CAT_CPC = {
   VIEW_TEST_RESULT_PAGE: 'ViewTestResultCatCPCPage',
   POST_DEBRIEF_HOLDING_PAGE: 'PostDebriefHoldingCatCPCPage',
 };
+
+type BasePageNames = {
+  [key in PageNameKeys]?: string;
+};
+type PageNameKeys =
+  'REKEY_UPLOAD_OUTCOME_PAGE' |
+  'BACK_TO_OFFICE_PAGE' |
+  'COMMUNICATION_PAGE' |
+  'DEBRIEF_PAGE' |
+  'HEALTH_DECLARATION_PAGE' |
+  'OFFICE_PAGE' |
+  'TEST_REPORT_PAGE' |
+  'WAITING_ROOM_PAGE' |
+  'WAITING_ROOM_TO_CAR_PAGE' |
+  'REKEY_REASON_PAGE' |
+  'PASS_FINALISATION_PAGE' |
+  'NON_PASS_FINALISATION_PAGE' |
+  'VIEW_TEST_RESULT_PAGE' |
+  'POST_DEBRIEF_HOLDING_PAGE' |
+  'VEHICLE_CHECKS_MODAL' |
+  'REVERSE_DIAGRAM_PAGE';
+
+/**
+ * Return correct page constant based upon category
+ * @param category
+ * @param pageName
+ */
+export function pageToNavigate(category: TestCategory, pageNameKey: PageNameKeys): string {
```

Comment:
can you rename this function to be more descriptive e.g. `getPageNameByCategoryAndKey()` ?

Reply:
renamed